mlir/lib/Target/LLVMIR/Dialect/OpenMP/OpenMPToLLVMIRTranslation.cpp

```diff
@@ -1000,11 +1000,39 @@ convertOmpWsLoop(Operation &opInst, llvm::IRBuilderBase &builder,
   return success();
 }
 
+/// Replace the region arguments of the parallel op (which correspond to private
+/// variables) with the actual private varibles they correspond to. This
+/// prepares the parallel op so that it matches what is expected by the
+/// OMPIRBuilder. Instead of editing the original op in-place, this function
+/// does the required changes to a cloned version which should then be erased by
+/// the caller.
+static omp::ParallelOp
+prepareOmpParallelForPrivatization(omp::ParallelOp opInst) {
```

Comment:
Where is this function invoked from? One probable place is in `convertOmpParallel`, but that line is commented out. 

I read the other discussion, and it makes sense that if `%arg` is used within the region, then `CodeExtractor` will not really consider it as `Input` and involve the appropriate privatization callback. But I am not able to connect where exactly are we modifying the parallel op?

Reply:
It was temporarily commented out. See the same comment I linked above for context.